packages/beacon-node/src/api/impl/beacon/state/utils.ts

```diff
@@ -111,7 +111,7 @@ export function toValidatorResponse(
 export function filterStateValidatorsByStatus(
   statuses: string[],
   state: BeaconStateAllForks,
```

Comment:
You can try to change this into `state: CachedBeaconStateAllForks` and access the epochCtx here. Try in the sense that I'm not sure if some callers do not have access to the epochCtx, but it would simply the arguments.

Reply:
> You can try to change this into `state: CachedBeaconStateAllForks` and access the epochCtx here. Try in the sense that I'm not sure if some callers do not have access to the epochCtx, but it would simply the arguments.

I would have a separate param `epochCache: EpochCache` and keep `state: BeaconStateAllForks` because some apis they pass in an epoch cache that’s different than the state they pass in. 

eg. the current logic of `getStateValidators(stateId, filters)` is to pass in the beacon state associated with `stateId`, but when looking up the validators, it uses the pubkey cache of the head state and filter after: https://github.com/ChainSafe/lodestar/blob/f5e2c3a40c9a01a29a4283d9f17a901ede413068/packages/beacon-node/src/api/impl/beacon/state/index.ts#L62

Reply:
> You can try to change this into `state: CachedBeaconStateAllForks` and access the epochCtx here. Try in the sense that I'm not sure if some callers do not have access to the epochCtx, but it would simply the arguments.

I would rather have a separate param `epochCache: EpochCache` and keep `state: BeaconStateAllForks` because the caller originally passes a particular beacon state and a pubkey cache of the head state. 

eg. `getStateValidators(stateId, filters)` passes the beacon state associated with `stateId`, and when looking up the validators, it uses the pubkey cache of the head state and filter the result after: https://github.com/ChainSafe/lodestar/blob/f5e2c3a40c9a01a29a4283d9f17a901ede413068/packages/beacon-node/src/api/impl/beacon/state/index.ts#L62

Having `state: CachedBeaconStateAllForks` here would force the caller to create a cached beacon state by combining the non head state with the epoch cache of the head state, which would look confusing

Reply:
may be call it a `headEpochCache` in the arg to make it more clear